provider/etcd.go

```diff
@@ -8,12 +10,16 @@ import (
 
 // Etcd holds configurations of the Etcd provider.
 type Etcd struct {
-	Kv `mapstructure:",squash"`
+	Kv  `mapstructure:",squash"`
+	TLS *KvTLS
```

Comment:
I created a branch that fixes your issue: https://github.com/containous/traefik/tree/fixes-etc-consul-tls